code/game/mecha/equipment/tools/tools.dm

```diff
@@ -1309,5 +1332,85 @@
 		do_after_cooldown()
 	return 1
 
+/obj/item/mecha_parts/mecha_equipment/tool/switchtool
+	name = "\improper Exosuit-Mounted Engineering Switchtool"
+	desc = "An exosuit-mounted Engineering switchtool. (Can be attached to: Any exosuit)"
+	icon_state = "mecha_rcd"	//needtochange
+	origin_tech = Tc_MATERIALS + "=3;" + Tc_POWERSTORAGE + "=2"
+	equip_cooldown = 10
+	energy_drain = 50
+	range = MELEE|RANGED
+	var/datum/global_iterator/pr_switchtool
+	var/obj/item/weapon/switchtool/engineering/mech/S
+
+/obj/item/mecha_parts/mecha_equipment/tool/switchtool/New()
+	..()
+	S = new(src)
+	pr_switchtool = new /datum/global_iterator/mecha_switchtool(list(src),0)
+	pr_switchtool.set_delay(equip_cooldown)
+	pr_switchtool.toggle()
+
+/obj/item/mecha_parts/mecha_equipment/tool/switchtool/Destroy()
```

Comment:
You're not clearing up that global iterator thing.

Reply:
I assumed it was taken care of automatically, since none of the other modules which use global iterators get rid of them in `Destroy()`.

Reply:
Mech code is damn ancient and shitty at that. It predates qdel so it has 0 care to give about reference clearing.